pkg/services/rpcsrv/server_test.go

```diff
@@ -2275,79 +2275,173 @@ func TestSubmitOracle(t *testing.T) {
+		t.Run("not a base64", submitNotaryRequest(t, true, neorpc.InvalidParamsCode, `"not-a-base64$"`))
+		t.Run("invalid request bytes", submitNotaryRequest(t, true, neorpc.InvalidParamsCode, `"not-a-request"`))
+		t.Run("invalid request", func(t *testing.T) {
+			mainTx := &transaction.Transaction{
+				Attributes:      []transaction.Attribute{{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 1}}},
+				Script:          []byte{byte(opcode.RET)},
+				ValidUntilBlock: 123,
+				Signers:         []transaction.Signer{{Account: util.Uint160{1, 5, 9}}},
+				Scripts: []transaction.Witness{{
+					InvocationScript:   []byte{1, 4, 7},
+					VerificationScript: []byte{3, 6, 9},
+				}},
+			}
+			fallbackTx := &transaction.Transaction{
+				Script:          []byte{byte(opcode.RET)},
+				ValidUntilBlock: 123,
+				Attributes: []transaction.Attribute{
+					{Type: transaction.NotValidBeforeT, Value: &transaction.NotValidBefore{Height: 123}},
+					{Type: transaction.ConflictsT, Value: &transaction.Conflicts{Hash: mainTx.Hash()}},
+					{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 0}},
+				},
+				Signers: []transaction.Signer{{Account: util.Uint160{1, 4, 7}}, {Account: util.Uint160{9, 8, 7}}},
+				Scripts: []transaction.Witness{
+					{InvocationScript: append([]byte{byte(opcode.PUSHDATA1), keys.SignatureLen}, make([]byte, keys.SignatureLen)...), VerificationScript: make([]byte, 0)},
+					{InvocationScript: []byte{1, 2, 3}, VerificationScript: []byte{1, 2, 3}}},
+			}
+			p := &payload.P2PNotaryRequest{
+				MainTransaction:     mainTx,
+				FallbackTransaction: fallbackTx,
+				Witness: transaction.Witness{
+					InvocationScript:   []byte{1, 2, 3},
+					VerificationScript: []byte{7, 8, 9},
+				},
+			}
+			nrBytes, err := p.Bytes()
+			require.NoError(t, err)
+			str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+			submitNotaryRequest(t, true, neorpc.ErrVerificationFailedCode, str)(t)
+		})
+		t.Run("valid request", func(t *testing.T) {
+			sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
+			notaryRequest = createValidNotaryRequest(chain, sender, 1)
+			nrBytes, err := notaryRequest.Bytes()
+			require.NoError(t, err)
+			str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+			submitNotaryRequest(t, false, 0, str)(t)
+		})
+	})
+
+	t.Run("getrawnotarypool", func(t *testing.T) {
+		t.Run("nonempty pool", func(t *testing.T) {
+			//get notary pool & check tx hashes
+			body := doRPCCallOverHTTP(rpcPool, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 1, len(actual.Hashes))
+			for actMain, actFallbacks := range actual.Hashes {
+				require.Equal(t, notaryRequest.MainTransaction.Hash(), actMain)
+				require.Equal(t, 1, len(actFallbacks))
```

Comment:
You've asked about several fallbacks for a single main transaction, have you managed to create this testcase? Currently it's only one fallback in test that is checked.

Reply:
Yep, sorry, forgot about it. Added.